server/src/api.rs

```diff
@@ -418,25 +435,257 @@ async fn delete_tasks_request(
     Ok(web::Json(res))
 }
 
+fn construct_filter(filter: &Filter) -> actix_web::Result<Condition> {
```

Comment:
This was a very critical part of the changes added here, allows for filtering requested tasks.